commandapi-plugin/src/main/java/dev/jorel/commandapi/config/ConfigGenerator.java

```diff
@@ -0,0 +1,80 @@
+package dev.jorel.commandapi.config;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+
+public class ConfigGenerator {
+
+	private final DefaultConfig defaultConfig;
+
+	private ConfigGenerator(DefaultConfig defaultConfig) {
+		this.defaultConfig = defaultConfig;
+	}
+
+	public static ConfigGenerator createNew(DefaultConfig defaultConfig) {
+		return new ConfigGenerator(defaultConfig);
+	}
+
+	public <T> void populateDefaultConfig(ConfigurationAdapter<T> adapter) {
+		for (Map.Entry<String, CommentedConfigOption<?>> commentedConfigOption : defaultConfig.getAllOptions().entrySet()) {
+			adapter.tryCreateSection(commentedConfigOption.getKey());
+			adapter.setValue(commentedConfigOption.getKey(), commentedConfigOption.getValue().option());
+			adapter.setComment(commentedConfigOption.getKey(), commentedConfigOption.getValue().comment());
+		}
+		adapter.complete();
+	}
```

Comment:
Note that since `tryCreateSection` no longer sets the comments, when the CommandAPI directory is created for the first time and this code is run, the `messages` section does not have a comment:

![image](https://github.com/user-attachments/assets/463f1369-91fa-408c-adc5-d304f10bb301)

So, basically this again https://github.com/JorelAli/CommandAPI/pull/596#discussion_r1770629791. I suppose you could add more code here to set the comments. Or, I think we could reduce the number of code paths that need to be tested and use `generateWithNewValues` with `existingConfig` being empty like I mentioned last time.

Automated unit tests would have caught this before merging :P